### PR TITLE
Dev sahil issue18

### PR DESCRIPTION
**Pull request by sahiilll**
3 Major Changes:

( The steps taken are documented under the links )
Root URL modification vital2/Vital-Architecture#32
on_web_server_startup script and network startup script / IPTABLES vital2/Vital-Architecture#18
RemoveCourse script https://github.com/vital2/ansible/issues/25

Kindly review it @reg1reg1 @treddington @ashishdas009

### CODE REVIEW

**Comment by treddington** [comment]
“Course Removal Script” is really course-reset-script - No?
Thomas Reddington
Industry Professor of Computer Science
treddington@nyu.edu
908-500-0097

**Comment by sahiilll** [comment]
https://github.com/vital2/vital-development/pull/53/commits/457aba5a83b3a5f573b964b8fe90e1be93e6b0bb'

Dear professor, Kindly refer to the commit above to review the changes for iptables. This does includes on_web_server_startup script and network_startup script which handles the rules when we set up new course. @treddington

**Comment by treddington** [comment]
The CN interfaces are not the same as prod. Prod is ok.

Thomas Reddington
Industry Professor of Computer Science
treddington@nyu.edu
908-500-0097

**Comment by treddington** [comment]
Different base VMs?

Thomas Reddington
Industry Professor of Computer Science
treddington@nyu.edu
908-500-0097

**Comment by treddington** [comment]
An error has occured. Please create a bug ticket on NYU Bugzilla <https://bugzilla.poly.edu/>. Including a description of the error and steps taken to re-create the bug will help us address this as effectively as possible. Alternatively you can send a mail to vital.nyu@gmail.com with Subject Header starting with "Error”.

On production CN

Thomas Reddington
Industry Professor of Computer Science
treddington@nyu.edu
908-500-0097

**Comment by sahiilll** [comment]
Dear professor,

I don't understand the doubt. Can we have a call some time later today ?

Thanks & Regards
Sahil

**Comment by reg1reg1** [comment]
@treddington Will look into it why we cannot start courses on dev, 3 possible main causes.
1) Xen1 not accepting requests to start vm's. Need to check this one out.
2)Could be a bridge failure issue (Issue 38), if this is the case will not persist across students.
3)Could be that course vm's or qcows were altered.
@ashishdas009 Can you confirm that you were able get vm's to run from the web interface on xen1? Since there is a load balancer, this might require multiple attempts?

File: vital_site/vital/views/security_view.py [hunk 1]
```diff
@@ -63,11 +64,7 @@ def register(request):
                 user.sftp_account = user.email[:user.email.find('@')]
                 user.sftp_pass = user.password  # workaround to set sftp account
                 user.set_password(user.password)  # hashes the password
-<<<<<<< HEAD
-                activation_code = str(uuid.uuid4()).replace('-', '')
```
Comment (reg1reg1) [inline vital_site/vital/views/security_view.py]: Not necessary,needs to be fixed.

Comment (sahiilll) [reply vital_site/vital/views/security_view.py]: No need to be fixed. Someone imported random lib and then replaced the way of generation of activation code. It should work fine.

Comment (sahiilll) [reply vital_site/vital/views/security_view.py]: Someone might have made this change on local repo earlier and got pushed through my branch

Comment (reg1reg1) [reply vital_site/vital/views/security_view.py]: Might be, the change head seems to be master. Needs to be fixed. The activation code you get on production is 6 digit number now and not uuid. 
image

**Pull request merged**